src/logic/safe/store/actions/createTransaction.ts

```diff
@@ -189,22 +190,42 @@ export class TxSender {
     })
   }
 
+  async canSignOffchain(state: AppReduxState): Promise<boolean> {
+    const { isFinalization, safeVersion } = this
+    const { account, name, smartContractWallet } = providerSelector(state)
+    let isSmart = smartContractWallet
+
+    // WalletConnect itself isn't a smart contract but the connected account can be
+    if (!isSmart && name.toUpperCase() === WALLET_PROVIDER.WALLETCONNECT) {
```

Comment:
Maybe we should always do this check?

The name returned from Onboard is not always the same as in our `WALLETS` enum. It will work fine for Wallet Connect as I think Onboard expects `"walletConnect"` in it's config and returns the `"WalletConnect"` name (it may be `"Wallet Connect"` though).

For desktop pairing, the config takes `"desktopPairing"` but the name is `"Gnosis Safe"` iirc.

Reply:
Done.
`desktopPairing` is a confusing name btw. Similarly to "deep link" we should not let our internal lingo trickle into the code.

Reply:
What would you suggest we call it? We need a name to enable/disable it.

Reply:
Since this is off-topic, let's continue on slack.